trace_decoder/src/observer.rs

```diff
@@ -0,0 +1,110 @@
+use std::collections::BTreeMap;
+use std::marker::PhantomData;
+
+use ethereum_types::H256;
+
+use crate::core::IntraBlockTries;
+use crate::typed_mpt::{ReceiptTrie, StorageTrie, TransactionTrie};
+
+/// Observer API for the trace decoder.
+/// Observer is used to collect various debugging and metadata info
+/// from the trace decoder run.
+pub trait Observer<StateTrieT: Clone> {
+    /// Collect tries after the transaction/batch execution.
+    ///
+    /// Passing the arguments one by one through reference, because
+    /// we don't want to clone argument tries in case they are not used in
+    /// observer.
+    fn collect_tries(
+        &mut self,
```

Comment:
@0xaatif I can add blanket implementation:
```
impl<T, StateTrieT> Observer<StateTrieT> for &mut T
where
    T: Observer<StateTrieT>,
{
    fn collect_tries(
        &mut self,
        _block: U256,
        _batch: usize,
        _state_trie: &StateTrieT,
        _storage: &BTreeMap<H256, StorageTrie>,
        _transaction_trie: &TransactionTrie,
        _receipt_trie: &ReceiptTrie,
    ) {
    }
}
```

but that still is not enough to remove `collect_tries` implementation for `DummyObserver`:
```
// impl<StateTrieT> Observer<StateTrieT> for DummyObserver<StateTrieT> 

85 |             &mut DummyObserver::new(),
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `trace_decoder::observer::Observer<trace_decoder::typed_mpt::StateMpt>` is not implemented for `trace_decoder::observer::DummyObserver<_>

```

because requirement is that `DummyObserver` already implements `Observer` (including `collect_tries` ). 
So I don't see at the moment what is the purpose of this blanket implementation?